metabase/metabase
# Let users create moderation requests and let admins create moderation reviews (on questions)

This PR adds the ability for normal users (not admins) to create moderation requests on saved questions. Once a BE bug has been dealt with admins will also be able to close moderation requests opened by users.

Steps to test:
1. If you don't have one on hand, create a non-admin user via http://localhost:3000/admin/people
2. Using your new user, navigate to a pre-existing saved question
3. Click on the header of the question to open up the details sidebar
4. Click on the "user action" dropdown button (real text tbd) and select one of the options
5. You should've been navigated to a form with a single textarea input. Fill it in or don't fill it in.
6. Click the blue save button
7. You should see a button in the details sidebar that shows how many open moderation requests exist for a given question (assuming this is a new question, there should now be one, the one you just created).
8. Click on the button and you should be navigated to an "open issues" view of the sidebar that shows the moderation request you just created.

Things to note:
1. I'm punting on adding real text so you may see placeholders like in the video below
2. I'm also punting on adding tests since we're still technically in a "prototype" phase on this project.

Here's a video showing a user creating a moderation request:

https://user-images.githubusercontent.com/13057258/118905690-14b4c900-b8d1-11eb-8d71-791752c8f62f.mov
